### PR TITLE
Replaces the trial of mana grass sprites + other misc. grass cleanup

### LINKED ISSUE #26449: Grass & dirt sprites need to be replaced / resprited
## Description
To be more specific, the offending sprites in question are ``Resources\Textures\Tiles\Planet\dirt.rsi`` and ``Resources\Textures\Tiles\Planet\Grass``.

image

Hilariously, these sprites are literally just straight up taken from Trials of Mana.
Outside of the obvious, these definitely do not fit the art style anyways (they're from an entirely different game, duh) and are pretty hueshifted & low in detail.

We have a few sprites in the repo to replace these already, namely ``grass.png``, ``grassdark.png``, and ``grasslight.png`` in the ``Resources\Textures\Tiles`` folder.. They are pretty noisy and likely won't look nearly as good, but should do the job alright.
If you can, you should probably add edge tiles to whatever one you decide to make the new "normal" grass, even if you suck at art and want to take this issue on. They are pretty important to making the grass biome look good.

### PR DESCRIPTION
**Pull request by Flareguy**
## About the PR
Fixes #26449.
Replaces the "planetmap" grass & dirt sprites from Trials of Mana with the grass sprites used in mowed astrograss + the other dirt sprite.
Also removes jungle grass variants, non-astro-grass tile stacks, & adds astro-dirt.

To compensate for the sprites being worse, grass now at least has trims.

## Why / Balance
I am fully aware this may be kind of overstepping for the freeze, so the PR can be scaled back if-needed.

Starting with the initial purpose of this PR, the default grass cannot be in upstream since it is from Trials of Mana. Between the 3 grass variants, the noisy grass seemed the best as a general replacement, since the other kind of grass was way too contrasted (see below image.)
Content Client_RR2iQIA2gT

Jungle grass was removed since I didn't really see much use for it. Stylistically, it's a pretty damn big departure from the other kind of grass, and it probably can't reasonably be used in planetmaps either, since it doesn't fit with the trees.

Astro-dirt is the most feature-like thing in this PR, but it's trivial to add and needed to exist. Mappers probably shouldn't be mapping indestructible tiles in to their maps.

## Media

ZlTq0v4QV6

Hey, it could be worse...

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Added astro-dirt to the protolathe. You can unlock it in the same research as all of the other astro-tiles.

### CODE REVIEW

**Comment by QuietlyWhisper** [comment]
Unironically like this new grass more

**Comment by EmoGarbage404** [comment]
would it be possible to make a variant of the existing grass texture with a hue similar to the ToM one? 

image

the decals specifically clash quite hard and would look better with the earthier green tone.

File: Resources/migration.yml [hunk 1]
```diff
@@ -335,3 +335,8 @@ ReinforcementRadioSyndicateMonkeyNukeops: ReinforcementRadioSyndicateAncestorNuk
 
 # 2024-05-01
 DrinkBottleGoldschlager: DrinkBottleGildlager
+
+# 2024-05-11
+FloorTileItemGrassJungle: FloorTileItemGrass
+FloorTileItemMowedAstroGrass: FloorTileItemAstroGrass
+FloorTileItemJungleAstroGrass: FloorTileItemAstroGrass
```
Comment (metalgearsloth) [inline Resources/migration.yml:342]: You need to use tile alises for tiles, not migrations.

**Comment by IamVelcroboy** [comment]
The jungle grass is pretty bad but the light/dark grass don't need to be removed. They look decent when you add decals to them. 

I'm worried that migrating all grass to this single variant is going to screw up all the existing decals mapped. I guess I'd have to test and see how this new one looks on existing maps.

File: Resources/migration.yml (hunk 1 above)
Comment (mirrorcult) [reply Resources/migration.yml:342]: these are the floor tile items, not the tiles themselves

**Comment by Ko4ergaPunk** [comment]
There is no reason to remove jungle grass. I use it for mapping.

Btw add astro-sand

**Comment by Flareguy** [comment]
This is too much for me to handle right now. I'm going to have to atomize this later.

**Pull request closed without merging**